blocktx/peer_handler.go

```diff
@@ -275,6 +276,20 @@ func (bs *PeerHandler) HandleBlock(wireMsg wire.Message, peer p2p.PeerI) error {
 		gocore.NewStat("blocktx").NewStat("HandleBlock").AddTime(start)
 	}()
 
+	primaryBlocktx, err := bs.store.PrimaryBlocktx(context.Background())
+	if err != nil {
+		return err
+	}
+
+	hostName, err := os.Hostname()
+	if err != nil {
+		return err
+	}
+
+	if primaryBlocktx != hostName {
+		return nil
```

Comment:
Here an Info log could be added that this `blocktx` is not primary so it will not do anything.